index.js

```diff
@@ -421,15 +430,13 @@ var RestModel = module.exports = Ember.Object.extend({
         resolve(processedCache);
       }
 
-      $.ajax({
-        url        : options.url,
-        type       : method,
-        data       : options.data,
-        headers    : options.headers || {},
-        beforeSend : self.getBeforeSend(options),
-        dataType   : 'json',
-        contentType: 'application/json'
-      }).then(function(data, responseText, jqXHR) {
+      $.ajax(ajaxOptions).then(function(data, responseText, jqXHR) {
+
+        if (processedCache && jqXHR.status === 304) {
+          // Cache is valid, do nothing.
+          return;
```

Comment:
I don't think so... Line 433 doesn't have a return either. So the return value from 437 doesn't go anywhere.